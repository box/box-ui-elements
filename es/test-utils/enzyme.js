function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map