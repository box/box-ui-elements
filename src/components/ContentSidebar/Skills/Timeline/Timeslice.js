/**
 * @flow
 * @file Timeline line component
 * @author Box
 */

import React from 'react';
import PlainButton from 'box-react-ui/lib/components/plain-button/PlainButton';
import './Timeslice.scss';

type Props = {
    start: number,
    duration: number,
    end?: number,
    index: number,
    onClick: Function,
    interactionTarget: string,
};

const LENGTH_TEXT_ITEMLINE = 290; // match with css
const MIN_WIDTH = 6; // Need at least some width to be clickable

const Timeslice = ({
    start,
    end,
    duration,
    onClick,
    index,
    interactionTarget,
}: Props) => {
    if (typeof start !== 'number' || !duration || start >= duration) {
        return null;
    }

    const barLength = LENGTH_TEXT_ITEMLINE;
    let startLeft = Math.round((start * barLength) / duration);
    const minEnding = startLeft + MIN_WIDTH; // Need at least some width to be clickable
    const ending =
        typeof end === 'number'
            ? Math.max(minEnding, (end * barLength) / duration)
            : minEnding;
    const endLeft = Math.round(Math.min(barLength, ending));
    let width = endLeft - startLeft;

    // If width is too small re-adjust the left position
    // to get to at least 6px wide for clickability
    if (width < MIN_WIDTH) {
        startLeft -= MIN_WIDTH - width;
        width = MIN_WIDTH;
    }

    return (
        <PlainButton
            type="button"
            className="be-timeline-time"
            style={{
                left: `${startLeft}px`,
                width: `${width}px`,
            }}
            onClick={() => onClick(index)}
            data-resin-target={interactionTarget}
        />
    );
};

export default Timeslice;
