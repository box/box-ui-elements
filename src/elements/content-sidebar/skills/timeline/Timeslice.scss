@import '../../../common/variables';

.be .btn-plain.be-timeline-time {
    position: absolute;
    height: 6px;
    background-color: $blue;
    border-radius: $bdl-border-radius-size-med;
    cursor: pointer;
    opacity: .6;

    &:focus,
    &:active,
    &:hover {
        opacity: 1;
    }
}
