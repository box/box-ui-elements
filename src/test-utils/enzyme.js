/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
