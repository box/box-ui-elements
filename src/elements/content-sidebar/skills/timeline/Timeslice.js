/**
 * @flow
 * @file Timeline line component
 * @author Box
 */

import React from 'react';
import PlainButton from '../../../../components/plain-button/PlainButton';
import './Timeslice.scss';

type Props = {
    duration: number,
    end?: number,
    index: number,
    interactionTarget: string,
    onClick: Function,
    start: number,
};

const LENGTH_TEXT_ITEMLINE = 290; // match with css
const MIN_WIDTH = 6; // Need at least some width to be clickable

const Timeslice = ({ start, end, duration, onClick, index, interactionTarget }: Props) => {
    if (typeof start !== 'number' || !duration || start >= duration) {
        return null;
    }

    const barLength = LENGTH_TEXT_ITEMLINE;
    let startLeft = Math.round((start * barLength) / duration);
    const minEnding = startLeft + MIN_WIDTH; // Need at least some width to be clickable
    const ending = typeof end === 'number' ? Math.max(minEnding, (end * barLength) / duration) : minEnding;
    const endLeft = Math.round(Math.min(barLength, ending));
    let width = endLeft - startLeft;

    // If width is too small re-adjust the left position
    // to get to at least 6px wide for clickability
    if (width < MIN_WIDTH) {
        startLeft -= MIN_WIDTH - width;
        width = MIN_WIDTH;
    }

    return (
        <PlainButton
            className="be-timeline-time"
            data-resin-target={interactionTarget}
            onClick={() => onClick(index)}
            style={{
                left: `${startLeft}px`,
                width: `${width}px`,
            }}
            type="button"
        />
    );
};

export default Timeslice;
