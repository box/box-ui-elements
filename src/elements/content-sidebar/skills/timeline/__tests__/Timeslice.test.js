import * as React from 'react';
import { shallow } from 'enzyme';
import Timeslice from '../Timeslice';

describe('elements/content-sidebar/Skills/Timeline/Timeslice', () => {
    test('should correctly render the time slice with no end', () => {
        const props = {
            start: 10,
            duration: 100,
            interactionTarget: 'foobar',
        };

        const wrapper = shallow(<Timeslice {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render the time slice with proper end', () => {
        const props = {
            start: 10,
            end: 20,
            duration: 100,
            interactionTarget: 'foobar',
        };

        const wrapper = shallow(<Timeslice {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render the time slice by re adjusting start to fit in', () => {
        const props = {
            start: 99,
            end: 100,
            duration: 100,
            interactionTarget: 'foobar',
        };

        const wrapper = shallow(<Timeslice {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
