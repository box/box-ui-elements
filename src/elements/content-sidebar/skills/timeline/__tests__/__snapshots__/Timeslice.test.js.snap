// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/Skills/Timeline/Timeslice should correctly render the time slice by re adjusting start to fit in 1`] = `
<PlainButton
  className="be-timeline-time"
  data-resin-target="foobar"
  onClick={[Function]}
  style={
    Object {
      "left": "284px",
      "width": "6px",
    }
  }
  type="button"
/>
`;

exports[`elements/content-sidebar/Skills/Timeline/Timeslice should correctly render the time slice with no end 1`] = `
<PlainButton
  className="be-timeline-time"
  data-resin-target="foobar"
  onClick={[Function]}
  style={
    Object {
      "left": "29px",
      "width": "6px",
    }
  }
  type="button"
/>
`;

exports[`elements/content-sidebar/Skills/Timeline/Timeslice should correctly render the time slice with proper end 1`] = `
<PlainButton
  className="be-timeline-time"
  data-resin-target="foobar"
  onClick={[Function]}
  style={
    Object {
      "left": "29px",
      "width": "29px",
    }
  }
  type="button"
/>
`;
